File: DecryptSubstitution_German.m
function y = DecryptSubstitution_German(x, k)
    n = size(x, 2);
    for i = 1 : 30
        kk(k(i) - 96) = i + 96;
    end;
    for i = 1 : n 
        xp = x(i) - 64;
        y(i) = kk(xp);
    end;
    y=char(y);

File: R/DecryptSubstitution_German.r
DecryptSubstitution_German <-
  function(ciphertext_ascii_dec, key_list)
  {
    k1 <- key_list - 96
    k2 <- seq(1, length(key_list)) + 96
    dectext_ascii_30 <- ciphertext_ascii_dec - 64
    
    for (i in 1:length(key_list))
      dectext_ascii_30[dectext_ascii_30 == k1[i]] = k2[i]
    
    return(dectext_ascii_30)
  }